### PR TITLE
LBAC-37: Encounter Access based on patient Location

### PR DESCRIPTION
**Pull request by VANKINEENITAWRUN**
## Description of what I changed

we use a global property to whether restrict on Encounter Location or Patient Location and use it for restrictions.
## Issue I worked on

see https://issues.openmrs.org/browse/LBAC-37

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/locationbasedaccess/aop/interceptor/EncounterServiceInterceptorAdvice.java [hunk 1]
```diff
@@ -88,7 +89,13 @@ public Object invoke(MethodInvocation invocation) throws Throwable {
     }
 
     public static Boolean doesEncounterBelongToGivenLocation(Encounter encounter, String sessionLocationUuid) {
-        Location location = encounter.getLocation();
+        Location location;
+        String encounterAccessType = Context.getAdministrationService().getGlobalProperty(LocationBasedAccessConstants.ENCOUNTER_RESTRICTION_TYPE_ENCOUNTER_LOCATION);
+        if(LocationBasedAccessConstants.ENCOUNTER_RESTRICTION_TYPE_PATIENT_LOCATION ==encounterAccessType){
```
Comment (suthagar23) [inline api/src/main/java/org/openmrs/module/locationbasedaccess/aop/interceptor/EncounterServiceInterceptorAdvice.java]: format this line

File: api/src/main/java/org/openmrs/module/locationbasedaccess/aop/interceptor/EncounterServiceInterceptorAdvice.java [hunk 2]
```diff
@@ -88,7 +89,13 @@ public Object invoke(MethodInvocation invocation) throws Throwable {
...
+            location = LocationUtils.getPersonLocation(encounter.getPatient());
+        } else{
```
Comment (suthagar23) [inline api/src/main/java/org/openmrs/module/locationbasedaccess/aop/interceptor/EncounterServiceInterceptorAdvice.java]: Do we need this `else` part? Can we assign `encounter.getLocation()` as default return to this function?

**Comment by dkayiwa** [comment]
@suthagar23 any more comments for this?

**Comment by suthagar23** [comment]
Thanks @VANKINEENITAWRUN

**Pull request merged**